lib/MusicBrainz/Server/Data/Utils.pm

```diff
@@ -400,14 +415,19 @@ sub remove_invalid_characters {
     # trim XML-invalid characters
     =~ s/[^\x09\x0A\x0D\x20-\x{D7FF}\x{E000}-\x{FFFD}\x{10000}-\x{10FFFF}]//gor
     # trim other undesirable characters:
-    # - zwsp
-    # - shy
     # - bom
-    # - Other, control
-    # - Other, surrogate
     # - Other, private use
     # - Noncharacters
-    =~ s/[\x{200B}\x{00AD}\x{FEFF}\p{Cc}\p{Co}${noncharacter_pattern}]//gr
+    =~ s/[\x{FEFF}\p{Co}${noncharacter_pattern}]//gr
```

Comment:
I was going to suggest removing `\p{Co}` here to fix MBS-10976, but I submitted that separately instead (https://github.com/metabrainz/musicbrainz-server/pull/1964) so that I could additionally fix the ISE and add tests.